sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCSVScan.scala

```diff
@@ -380,7 +380,7 @@ class CSVPartitionReader(
     val csvOpts = buildCsvOptions(parsedOptions, readDataSchema, hasHeader)
     val dataSize = dataBufferer.getLength
     try {
-      withResource(dataBufferer.getBufferAndRelease) { dataBuffer =>
+      RmmRapidsRetryIterator.withRetryNoSplit(dataBufferer.getBufferAndRelease) { dataBuffer =>
```

Comment:
Will it keep retrying work when getting RetryOOM exception ?

Reply:
No not really. In the JNI state tracking code we have a limit of 500 retry operations for each task.

https://github.com/NVIDIA/spark-rapids-jni/blob/4f122d7a5ec655e17c159812709c75f1cbaa977c/src/main/cpp/src/SparkResourceAdaptorJni.cpp#L569

Also if all of the tasks are stuck on retry, then the JNI code will throw an exception asking the task to split the input and retry with smaller data. This does not cover split so if a split is thrown the task will fail and the entire task will be retried based off of how many times Spark is configured to retry.